app/models/grda_warehouse/tasks/identify_duplicates.rb

```diff
@@ -44,55 +45,67 @@ def identify_duplicates
+      # source client should be merged into the destination client.
+      # If the source client does not exist in any of the pairs, a new destination client should be created for it.
+      destination_clients_by_id = GrdaWarehouse::Hud::Client.where(id: found_ids.to_a).index_by(&:id)
+
       unprocessed.find_in_batches do |batch|
         matched_ids = []
         destination_client_updates = []
         new_destination_clients = []
         new_warehouse_clients = {}
         source_client_ids_with_new_destination_clients = []
+
         batch.each do |client|
-          match_id = check_for_obvious_match(client)
-          if match_id.present?
-            matched += 1
-            matched_ids << match_id
-            destination_client = get_destination_client_by_id(match_id)
+          destination_client = nil
+          if found_ids.include?(client.id)
+            matched_pair = more_than_one_match.find { |pair| pair.include?(client.id) }
+            destination_id = (matched_pair - [client.id]).first # We don't know the order of the pair
+            matched_ids << destination_id
+            new_warehouse_clients[client.id] = GrdaWarehouse::WarehouseClient.new(
+              id_in_source: client.personal_id,
+              source_id: client.id,
+              destination_id: destination_id,
+              data_source_id: client.data_source_id,
+            )
 
+            destination_client = destination_clients_by_id[destination_id]
             # Set SSN & DOB if we have it in the incoming client, but not in the destination
-            should_save = false
-            if client.DOB.present? && destination_client[:DOB].blank?
-              destination_client[:DOB] = client.DOB
-              should_save = true
-            end
-            if client.SSN.present? && destination_client[:SSN].blank?
-              destination_client[:SSN] = client.SSN
-              should_save = true
-            end
-            if should_save
-              # set non-nullable fields, these aren't used because of the column limitation on import
-              # but Postgres complains if they aren't there
-              destination_client[:PersonalID] = client.PersonalID
-              destination_client[:data_source_id] = @dnd_warehouse_data_source.id
-              destination_client_updates << destination_client
-            end
+            destination_client.dob = client.dob if client.dob.present? && destination_client.dob.blank?
+            destination_client.ssn = client.ssn if client.ssn.present? && destination_client.ssn.blank?
+            # set non-nullable fields, these aren't used because of the column limitation on import
+            # but Postgres complains if they aren't there
+            destination_client.personal_id = client.personal_id
+            destination_client.data_source_id = @dnd_warehouse_data_source.id
+            destination_client_updates << destination_client
```

Comment:
we're not always updating the record right?
```suggestion
            destination_client_updates << destination_client if destination_client.changed?
```